### ISSUE TITLE
Support DNSMadeEasy API 2.0

### ISSUE THREAD

**Issue by raravena80** [fog/fog#3884]
I don't think this is supported yet.

https://www.dnsmadeeasy.com/integration/pdf/API-Docv2.pdf

The new endpoint for the API is 

```
    if sandbox
      self.base_uri = 'https://api.sandbox.dnsmadeeasy.com/V2.0'
    else
      self.base_uri = 'https://api.dnsmadeeasy.com/V2.0'
    end
```

It's not compatible with this:

```
        def initialize(options={})

          @dnsmadeeasy_api_key = options[:dnsmadeeasy_api_key]
          @dnsmadeeasy_secret_key = options[:dnsmadeeasy_secret_key]
          @connection_options = options[:connection_options] || {}
          @host       = options[:host]        || 'api.dnsmadeeasy.com'
          @persistent = options.fetch(:persistent, true)
          @port       = options[:port]        || 80 #443 Not yet
          @scheme     = options[:scheme]      || 'http' #'https Not yet
          @connection = Fog::XML::Connection.new("#{@scheme}://#{@host}:#{@port}", @persistent, @connection_options)
        end
```

The `@connection` variable would have to change to something like this 

```
          @connection = Fog::XML::Connection.new("#{@scheme}://#{@host}:#{@port}/V2.0", @persistent, @connection_options)
```

**Comment by geemus** [comment]
From the sounds of it, you found out it will take more extensive changes than this? Happy to discuss if that would help.

**Comment by raravena80** [comment]
@geemus I created a fog-dnsmadeeasy with support for the 2.0 API.  How do we get it into the mainstream ?  https://github.com/raravena80/fog-dnsmadeeasy.  I think we would have to create https://github.com/fog/fog-dnsmadeeasy. Thx

**Comment by plribeiro3000** [comment]
@raravena80 The first step is to ensure it is following our conventions.
Then we need to add it to Fog and remove the old support within the main repo.

Have you seen [this](https://github.com/fog/fog/wiki/Create-New-Provider-from-Scratch) wiki page explaining how to create a new provider?

**Comment by raravena80** [comment]
@plribeiro3000 I basically followed the template for fog-dnsimple .  Do we need to support ruby 1.8.7 and ruby 1.9.3 ?  Also do we need minitest, I already have a set of tests under ./test

Thx

**Comment by geemus** [comment]
@raravena80 you should use the tests/testing that works for you. Ideally you would support 1.8.7/1.9.3, but it is more a "best effort" than requirement thing (which we hope to be done with sooner, rather than later).

**Comment by raravena80** [comment]
@geemus can we push this upstream ?

**Comment by geemus** [comment]
@raravena80 sure. Can you add me as an admin on the repo? Once that is done I can move it over and get things setup. Thanks!

**Comment by geemus** [comment]
I have access now, but I can't transfer it without admin level rights, could you escalate my permissions so I can do that? Thanks!

**Comment by raravena80** [comment]
@geemus is this fixed?

**Comment by geemus** [comment]
Hey, thanks for checking in. I hadn't realized that the permissions had been updated for me, I'll transfer now.

**Comment by geemus** [comment]
Er, never mind, it does look like this is fixed. I must have already done it and forgotten. Just let me know if you need anything else here.

**Comment by raravena80** [comment]
@geemus thx!
